### PR TITLE
Lint short description text

### PR DESCRIPTION
**Pull request by ddbeck**
This adds some JS to check short descriptions against our writing guidelines https://github.com/mdn/short-descriptions/blob/master/writing-short-descriptions.md#writing-short-descriptions.

A few notes to help you review this:

* This supersedes https://github.com/mdn/data/pull/297. There were several changes since the previous pull request. If you looked at the previous PR, it might help to step through the commits on this PR, rather than reviewing the thing from scratch.
* I'm pretty new to JavaScript and Node.js. I'd welcome any comments on style, idiom, etc.
* Once we get actual short descriptions into the data itself, we'll probably want to switch to checking the short description text in the repo, instead of fetching the MDN page (and use a different script to fetch the pages and update the text in the repo). I think this is a first iteration, not necessarily the final say on how to check these things.

Let me know if you've got any questions. Thanks, @wbamberg!

### CODE REVIEW

File: package.json
Comment (wbamberg) [inline package.json]: I get a warning here, it seems like it wants: CC-BY-SA-2.5 ?

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @ddbeck ! I haven't looked at the code in fine detail, but I have read through it, and it all makes sense, is clearly written, and it seems to work fine.

The main comment I have is that I think we should distinguish 2 types of "violations":

* ones that should be fixed in the Wiki page
* ones that should be fixed automatically as part of the process of importing prose from the Wiki, but left intact in the Wiki page

In particular, regarding `title` attributes, especially those inserted by the `cssxref` macro: we might at some point have consensus to remove these in the Wiki, but don't yet, and so I think we should strip them out in the process of importing them, and not fix them in the Wiki. On the other hand, length violations we should fix in the Wiki.

I'm not quite sure what that means for the linter, at this stage. In a future version I guess it means it (or some other script) will try to fix things like `title` attributes.

I'm also not sure yet which things are "things like `title` attributes" - one pragmatic way to look at it is to say: if we can't reliably fix things automatically on import, we probably should fix them in the Wiki. So for instance we can't automatically shorten summaries without garbling them, but we can rautomatically emove `title` and probably `class` attributes.

Another way to look at it is: will fixing them in the Wiki improve the Wiki content as well as the short-descriptions content? So if we think summaries <= 180 characters long are good for the Wiki, we should fix them in the Wiki. If some other guidelines make sense for the npm short-descriptions repo but not for the Wiki, maybe we shouldn't.

Does that make sense?

Other comments:

* style guide stuff: I don't think we have a style guide for code like this. I'd say in general, use the AirBnb style guide. The only thing here that leaps out at me is semicolons at the ends of lines, which we should have (https://github.com/airbnb/javascript#semicolons).

* test cases: one thing I found with this script is that it's a little hard to see which rules are being applied - they're a bit buried down at lines 123-142. You could describe the list of tests in a comment at the top but then of course it gets out of sync with the code. I'm hand-waving here, but perhaps you could consider having a standard form of "test function", with a common input (e.g. the summary contents) and output (e.g. success/failure). You could declare a const array of these things at the top, that then describes the tests that will be applied. Then `checkSummary` can walk through the array applying each test.

All that said, the only thing here blocking this from being merged, IMO, is the semicolons thing. If you wanted to address the others before merging that would be fine, but I'm also happy to keep them as future considerations.

File: package.json
Comment (ddbeck) [reply package.json]: Fixed!

**Comment by ddbeck** [comment]
> [insert discussion on wiki-fixable/source-fixable stuff] Does that make sense?

Yes, it does! I think maybe the right thing here is to leave checks here and remove them when we have an import script that deals with that particular issue.

> I'd say in general, use the AirBnb style guide.

I've fixed the semicolons. I tried eslint with the Airbnb style and there are a few issues. Rather than overloading this review, I'll start a separate PR (after this one) to add a linter and clean up the issues it raises.

> I'm hand-waving here, but perhaps you could consider having a standard form of "test function"

I sorta started this here. There's now a common function signature for the checks, but it's a little awkward to put the list of checks at the top of the file—JavaScript doesn't like references to undefined functions defined with `const`. Anyway, there's now a more concise list of checks just before they're invoked, so I think it helps a little. Let me know if you have any more ideas about this.

**Comment by ddbeck** [comment]
Hi @wbamberg, if you could take a look at this some time this week, I think we can wrap it up before the end of the sprint

**Review by wbamberg — APPROVED** [review]
Thanks @ddbeck ! This all makes sense to me :).

**Pull request merged**
